docs/proposals/external_providers_grpc.md

```diff
@@ -54,3 +54,22 @@ https://github.com/gardener/machine-controller-manager/tree/grpc-driver
 
 Refer GRPC based implementation for AWS client:
 https://github.com/ggaurav10/aws-driver-grpc
+
+## Alternatives:
+### Cloud Provider as GRPC server:
+One approach would be to make MCM as a GRPC client and Cloud Provider as a GRPC server.
+Here the Cloud Provider will provide the following GRPC services:
+1. `Create()` creates a VM
+1. `Delete()` deletes a VM
+1. `List()` lists VMs
+
+In this approach also, a GRPC stream would be required so that the CP can request for machine class and secret from MCM using the functions `GetMachineClass()` and `GetSecret()`.
+A disadvantage of this approach would be that each Cloud Provider will be opening a different port in the system which increases its attack surface.
+
+### Cloud Provider as Webhook:
+Another approach would be to use webhooks instead of CP code running in the system as another pod or a sidecar. This would require the CP to contact API server to read machine class and secret. A better approach might be to implement a controller instead of webhooks. This brings us to next possible approach.
+
+### Cloud Provider as a controller:
+Idea is to separate machine controller from rest of the controllers (machine set, machine deployment and safety controllers) and integrate it with the Cloud Provider's code. Basically, each CP also implements its own machine controller which will have access to all the information (machine class and secret details) required to do Cloud operations.
+This approach is currently under discussion in the SIG-cluster-lifecycle.
+This approach was not preferred in our implementation because this decreases the ease with which support for a new CP can be added to MCM.
```

Comment:
s/This approach was not preferred in our implementation/This approach is not invalidated by our implementation. Our implementation gives another (potentially, easier) way to achieve the same goals./